lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -415,10 +415,12 @@ function _campaign_resource_reportback($nid, $values) {
     return $rb_info['rbid'];
   }
   else {
-    // Increment transaction id and log that the request has been received with new transaction id.
-    $incremented_transaction_id = dosomething_api_increment_transaction_id($headers['X-Request-Id']);
+    if ($headers['X-Request-Id']) {
```

Comment:
What is this check for? Above you check if it is set, but if it is not, you check if it is set again?